### PR TITLE
JDBC better error handling

### PR DESCRIPTION
**Pull request by Jens-H**
This PR adds better error handling in the JNI part by returning to JVM immediately after creating a Java error. Also the the Java part should handle cases better, where a c++ struct was created but an error got raised during the further processing.

There is one question left: Can one expect that the constructor of DuckDB (and Connection) handles exceptions himself and does not leak memory? E.g. here `auto db = new DuckDB(database, &config);`

### CODE REVIEW

File: tools/jdbc/src/jni/duckdb_java.cpp [hunk 1]
```diff
@@ -206,6 +206,7 @@ JNIEXPORT jobject JNICALL Java_org_duckdb_DuckDBNative_duckdb_1jdbc_1startup(JNI
 		return env->NewDirectByteBuffer(db, 0);
 	} catch (exception &e) {
 		env->ThrowNew(J_SQLException, e.what());
+		return 0;
```
Comment (Mytherin) [inline tools/jdbc/src/jni/duckdb_java.cpp:209]: Should this not be `return nullptr`?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - one minor comment.

File: tools/jdbc/src/jni/duckdb_java.cpp (hunk 1 above)
Comment (Jens-H) [reply tools/jdbc/src/jni/duckdb_java.cpp]: I changed all return 0 occurrences to return nullptr.

**Comment by Mytherin** [comment]
Thanks for the updates! LGTM

**Pull request merged**